### DISCUSSION TITLE
Am I filtering wrong?

### DISCUSSION THREAD

**Discussion by kb6673** [rappasoft/laravel-livewire-tables#274] [Q&A] [answered]
I've got a fairly simple table that looks like this

```
  Column::make( 'Address' )->searchable()->sortable()->format( function ( $value, $column, $row ) {
                    return $row->inline_address;
                } )->asHtml(),
```

The main thing is here is we are collecting some separate address fields into one column `$row->inline_address` with this function 

```
    public function getInlineAddressAttribute() {
        $address = collect( [
            $this->address_line_1,
            $this->address_line_2,
            $this->town_city,
            $this->county,
            $this->postcode,
            $this->country
        ] )->reject( function ( $name ) {
            return empty( $name );
        } );

        return implode( ', ', $address->toArray() );
    }
```

![image](https://user-images.githubusercontent.com/1001484/117300828-de5b6080-ae71-11eb-987e-33c9c788d93c.png)

Into

![image](https://user-images.githubusercontent.com/1001484/117300856-e4514180-ae71-11eb-8807-08f6f25583f7.png)

However, during the search, it is trying to search a non-existent field called `address`

![image](https://user-images.githubusercontent.com/1001484/117301110-34c89f00-ae72-11eb-8fcf-0cda4c20efca.png)

Using this `query()` function I'm not asking it to search a `address` field

```
public function query(): Builder {
        return Company::query()->when( 
            $this->getFilter( 'search' ), fn( $query, $term ) => $query
                ->where( 'name', 'like', '%' . $term . '%' )
                ->orWhere( 'address_line_1', 'like', '%' . $term . '%' )
                ->orWhere( 'address_line_2', 'like', '%' . $term . '%' )
                ->orWhere( 'town_city', 'like', '%' . $term . '%' )
                ->orWhere( 'county', 'like', '%' . $term . '%' )
                ->orWhere( 'postcode', 'like', '%' . $term . '%' )
                ->orWhere( 'phone_number', 'like', '%' . $term . '%' ) 
        );
    }
```

Am I missing something obvious?

---
Full component

```
class CompaniesTable extends DataTableComponent {

    public function columns(): array {
        return [
            Column::make( '' )->format( function ( $value ) {
                return $value->avatar;
            } )->asHtml(),
            Column::make( 'Name' )->searchable()->sortable(),
            Column::make( 'Address' )->searchable()->sortable()->format( function ( $value, $column, $row ) {
                    return $row->inline_address;
                } )->asHtml(),
            Column::make( 'Phone Number' )->searchable()->sortable(),
            Column::make( 'Applicants' )->sortable()->format( function ( $value, $column, $row ) {
                return $row->applicant_count;
            } )->asHtml(),
        ];
    }

    public function getTableRowUrl( $row ): string {
        return route( 'companies.show', $row );
    }

    public function query(): Builder {
        return Company::query()->when( $this->getFilter( 'search' ), fn( $query, $term ) => $query->where( 'name', 'like', '%' . $term . '%' )->orWhere( 'address_line_1', 'like', '%' . $term . '%' )->orWhere( 'address_line_2', 'like', '%' . $term . '%' )->orWhere( 'town_city', 'like', '%' . $term . '%' )->orWhere( 'county', 'like', '%' . $term . '%' )->orWhere( 'postcode', 'like', '%' . $term . '%' )->orWhere( 'phone_number', 'like', '%' . $term . '%' ) );
    }
}
```

**Comment by rappasoft** [comment, marked as answer]
`address` comes from the column because you didn't specify.

`Column::make( 'Address' )` will assume a column name of `address` unless you pass a second parameter.

Search uses that column, you can override the search functionality with a closure: https://github.com/rappasoft/laravel-livewire-tables/wiki/Using-the-built-in-searching

I'm assuming you'll get the same error if you try to sort, because address doesn't exist but you're not telling it what to use otherwise.

**Reply by rappasoft** [reply]
Also, you don't use both `searchable()` on columns and a search filter in the query, it's one or the other. Since you have columns that are magic model properties, I would get rid of `searchable()` on the columns and specify all of your search criteria in the query.

**Comment by kb6673** [comment]
🎆 Removing `searchable()` from `Column::make` and keeping the query builder solves it. Obvious now you've explained it. :)

Much appreciated.
